### ISSUE TITLE
BlenderBIM: Problem when creating a new IfcElementType

### ISSUE THREAD

**Issue by vuicBABS** [IfcOpenShell/IfcOpenShell#1961]
Hi guys,

A problem occurs when creating a new IfcElementType based on an existing object or existing mesh. Steps to follow to reproduce the error:
1. Open a new file and create an Ifc Project
2. Import an object (append method) or mesh from a separate file (e.g. a door).
3. Under Object Properties --> IfcClass, assign a new class (Products: IfcElementType; Class: IfcDoorType; Predefined Type: DOOR; Context: Model/Body/MODEL_VIEW)

the error message: 

```
Python: Traceback (most recent call last):
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\bim\module\root\operator.py", line 129, in execute
    return IfcStore.execute_ifc_operator(self, context)
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\bim\ifc.py", line 278, in execute_ifc_operator
    result = getattr(operator, "_execute")(context)
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\bim\module\root\operator.py", line 140, in _execute
    self.assign_class(context, obj)
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\bim\module\root\operator.py", line 162, in assign_class
    blenderbim.core.geometry.add_representation(
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\core\geometry.py", line 59, in add_representation
    ifc.run(
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\tool\ifc.py", line 28, in run
    return ifcopenshell.api.run(command, IfcStore.get_file(), **kwargs)
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\libs\site\packages\ifcopenshell\api\__init__.py", line 46, in run
    result = usecase_class(ifc_file, **settings).execute()
  File "xxx\Blender Foundation\Blender\3.0\scripts\addons\blenderbim\libs\site\packages\ifcopenshell\api\style\assign_representation_styles.py", line 23, in execute
    name = style.Name
AttributeError: 'NoneType' object has no attribute 'Name'

location: <unknown location>:-1
```

**Comment by Moult** [comment]
Hey hey, I don't think this is necessarily a bug.

It sounds as though you are using the Blender append method (File->Append) to add an object from one IFC project to your current Blender session. This is not a supported workflow, as File->Append only works for pure Blender objects. Same for copy pasting between Blender sessions.

If you want to bring over data from one IFC project to another, you've got two options:

 1. In a Blender session of Project A, select your object, then in the IFC Class panel choose "Unlink Object". This breaks the Blender<->IFC link, so that object turns into a dumb Blender object. Then you can copy paste or File->Append to take that dumb object as you wish, and recreate it as a fresh IFC element in a new project.
 2. Use the IFC Project Library feature. This is like the IFC equivalent of Blender's Append feature. In the Scene properties under IFC Project Library browse to another IFC file, then you should be able to append type elements. Note that right now only appending types is supported, not actual occurrences.

Hope it helps, or have I misunderstood?

**Comment by vuicBABS** [comment]
Hi Dion,

thanks for the answer. I had tried the first variant (append blend file without ifc) and I was getting an error after assigning an ElementType (e.g. IfcDoorType) to the imported blender object. There was no error when assigning an Element (e.g. IfcDoor).
I have just tried again with 2 ElementType (IfcDoorType and IfcFilterType). Everything seems to work.
Sorry for the inconvenience and thanks for support.

Sincerely
